go-controller/pkg/ovn/pods.go

```diff
@@ -81,6 +81,27 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 	portInfo, err := oc.logicalPortCache.get(logicalPort)
 	if err != nil {
 		klog.Errorf(err.Error())
+		// If ovnkube-master restarts, it is also possible the Pod's logical switch port
+		// is not readded into the cache. Delete logical switch port anyway.
+		cmd, err := oc.ovnNBClient.LSPDel(logicalPort)
+		if err == nil {
+			if err = oc.ovnNBClient.Execute(cmd); err != nil {
+				klog.Errorf("Error while deleting logical port: %s, %v", logicalPort, err)
+			}
+		} else if err != goovn.ErrorNotFound {
+			klog.Errorf(err.Error())
+		}
```

Comment:
This same code is duplicated below
https://github.com/ovn-org/ovn-kubernetes/pull/1737/files#diff-41d43681ed1065378522fb129460853fR121-R128

maybe we can start to encapsulate it later in their own methods for better readability

Reply:
Thanks. Moved it to a utility function OvnNBLSPDel().